router_test.ts

```diff
@@ -0,0 +1,293 @@
+import { createRouter } from "./router.ts";
+import { describe, expect, fn, it } from "./dev_deps.ts";
+import { Status, STATUS_TEXT } from "./deps.ts";
+
+const describeTests = describe("createRouter");
+
+it(
+  describeTests,
+  "should return 404 when url path does not match",
+  async () => {
+    const router = createRouter({});
+    const res = await router(new Request("http://localhost/"));
+
+    expect(res).toEqualResponse(
+      new Response(null, {
+        status: Status.NotFound,
+        statusText: STATUS_TEXT[Status.NotFound],
+      }),
+    );
+  },
+);
+
+it(
+  describeTests,
+  `should return 404 when the handler is not defined`,
+  async () => {
+    const router = createRouter({
+      "/": {},
+    });
+    const res = await router(
+      new Request("http://localhost/"),
+    );
+
+    expect(res).toEqualResponse(
+      new Response(null, {
+        status: Status.NotFound,
+        statusText: STATUS_TEXT[Status.NotFound],
+      }),
+    );
+  },
+);
+
+it(
+  describeTests,
+  "should return 405 when method is not exists",
+  async () => {
+    const router = createRouter({
+      "/": {
+        "GET": () => new Response("hello"),
```

Comment:
Codacy has a fix for the issue: [Replace `··"GET"` with `GET`](https://app.codacy.com/gh/httpland/http-router/pullRequest?prid=9963312)
```suggestion
      GET: () => new Response("hello"),
```